### PR TITLE
Docker: able to not create the profile by passing env variable

### PR DESCRIPTION
**Pull request by unkcpz**

### CODE REVIEW

File: .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh [hunk 1]
```diff
@@ -5,24 +5,21 @@
...
-fi
 
-# Setup AiiDA profile if needed.
-if [[ ${NEED_SETUP_PROFILE} == true ]]; then
+    export AIIDA_PROFILE_NAME=default
+    export AIIDA_USER_EMAIL=aiida@localhost
+    export AIIDA_USER_FIRST_NAME=Giuseppe
+    export AIIDA_USER_LAST_NAME=Verdi
+    export AIIDA_USER_INSTITUTION=Khedivial
+    export AIIDA_PROFILE_PATH=/aiida/assets/config-quick-setup.yaml
```
Comment (sphuber) [inline .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh:22]: Should these also maybe just set these defaults only if not already explicitly set?
e.g.
```
export AIIDA_PROFILE_NAME="${AIIDA_PROFILE_NAME:-default}"
```

File: .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh [hunk 2]
```diff
@@ -5,24 +5,21 @@
...
-export AIIDA_USER_EMAIL=aiida@localhost
-export AIIDA_USER_FIRST_NAME=Giuseppe
-export AIIDA_USER_LAST_NAME=Verdi
-export AIIDA_USER_INSTITUTION=Khedivial
-export AIIDA_PROFILE_PATH=/aiida/assets/config-quick-setup.yaml
+# If the environment variable `SETUP_DEFAULT_AIIDA_PROFILE` is not set, set it to `true`.
+if [ -z ${SETUP_DEFAULT_AIIDA_PROFILE+x} ]; then
+    # Configure AiiDA.
+    export SETUP_DEFAULT_AIIDA_PROFILE=true;
+fi
```
Comment (sphuber) [inline .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh:12]:
```suggestion
export SETUP_DEFAULT_AIIDA_PROFILE="${SETUP_DEFAULT_AIIDA_PROFILE:-true}"
```

**Review by sphuber — CHANGES_REQUESTED** [review]

File: .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh (hunk 1 above)
Comment (sphuber) [reply .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh:22]: Also, maybe update the docs to reflect that these variables can be configured

Comment (unkcpz) [reply .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh]: Yes, this looks much simpler, and clear.

File: .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh [hunk 3]
```diff
@@ -5,34 +5,19 @@
...
-        --first-name "${AIIDA_USER_FIRST_NAME}"      \
-        --last-name "${AIIDA_USER_LAST_NAME}"        \
-        --institution "${AIIDA_USER_INSTITUTION}"    \
-        --config "${AIIDA_PROFILE_PATH}"
+        --profile "${AIIDA_PROFILE_NAME:-default}"            \
+        --email "${AIIDA_USER_EMAIL:-aiida@localhost}"                \
+        --first-name "${AIIDA_USER_FIRST_NAME:-Giuseppe}"      \
+        --last-name "${AIIDA_USER_LAST_NAME:-Verdi}"        \
+        --institution "${AIIDA_USER_INSTITUTION:-Khedivial}"    \
+        --config /aiida/assets/config-quick-setup.yaml
```
Comment (sphuber) [inline .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh:20]: Why not make this also configurable?

Comment (unkcpz) [reply .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh:20]: You mean the last one? It is the file for config (https://github.com/aiidateam/aiida-core/blob/main/.docker/aiida-core-base/s6-assets/config-quick-setup.yam) which contains things that I think usually fixed for a regular use of setting. 

It is copy to the container by https://github.com/aiidateam/aiida-core/blob/6ee278cebe8fb58cd6e69517d678b97570d0d661/.docker/aiida-core-base/Dockerfile#L166, so the file path makes no sense to be configurable, I assume.

Comment (sphuber) [reply .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh:20]: Yes, but the user may add a custom config file in a mounted volume and thereby customize the profile parameters, right?

Comment (unkcpz) [reply .docker/aiida-core-base/s6-assets/init/aiida-prepare.sh]: You are right. I added it back and add a note to the Doc.

File: docs/source/intro/run_docker.rst [hunk 4]
```diff
@@ -33,6 +33,20 @@ This image contains a fully pre-configured AiiDA environment which makes it part
...
+      - ``AIIDA_USER_FIRST_NAME`` the first name of the user to be created (default: ``Giuseppe``)
+      - ``AIIDA_USER_LAST_NAME`` the last name of the user to be created (default: ``Verdi``)
+      - ``AIIDA_USER_INSTITUTION`` the institution of the user to be created (default: ``Khedivial``)
+      - ``AIIDA_CONFIG_FILE`` the path to the AiiDA configuration file (default: ``/aiida/assets/config-quick-setup.yaml``).
+
+      You can set these environment variables when starting the container with the ``-e`` option.
+
+      If you don't want the profile to be created automatically, you can set the ``SETUP_DEFAULT_AIIDA_PROFILE`` environment variable to ``false``.
+
+      Please not that ``AIIDA_CONFIG_FILE`` point to the path inside the container, if you want to use a custom configuration file, you need to mount it from host path to the container path.
```
Comment (sphuber) [inline docs/source/intro/run_docker.rst:48]:
```suggestion
      By default, an AiiDA profile is automatically set up inside the container.
      To disable this default profile being created, set the ``SETUP_DEFAULT_AIIDA_PROFILE`` environment variable to ``false``.

      The following environment variables can be set to configure the default AiiDA profile:
      - ``AIIDA_PROFILE_NAME``: the name of the profile to be created (default: ``default``)
      - ``AIIDA_USER_EMAIL``: the email of the default user to be created (default: ``aiida@localhost``)
      - ``AIIDA_USER_FIRST_NAME``: the first name of the default user to be created (default: ``Giuseppe``)
      - ``AIIDA_USER_LAST_NAME``: the last name of the default user to be created (default: ``Verdi``)
      - ``AIIDA_USER_INSTITUTION``: the institution of the default user to be created (default: ``Khedivial``)
      - ``AIIDA_CONFIG_FILE``: the path to the AiiDA configuration file used for other profile configuration parameters (default: ``/aiida/assets/config-quick-setup.yaml``).

      These environment variables can be set when starting the container with the ``-e`` option.

      Please note that the ``AIIDA_CONFIG_FILE`` variable points to a path inside the container.
      Therefore, if you want to use a custom configuration file, it needs to be mounted from the host path to the container path.
```

**Review by sphuber — APPROVED** [review]

**Comment by sphuber** [comment]
Thanks @unkcpz

**Pull request merged**